### ISSUE TITLE
Add weight parameter for training C-SVC SVMs on unbalanced data

### ISSUE THREAD

**Issue by dbogdanov** [MTG/gaia#18]
Libsvm supports working with unbalanced data via adding class weights. Add a new parameter for SVM configuration which will allow to configure weights for each class. Currently the weights are [unassigned by default](https://github.com/MTG/gaia/blob/1f5921466174d92ace279a13dde4146c725fc95f/src/algorithms/svmtrain.cpp#L117), and there is not way to configure them. 

Make sure that cross-validation is not affected by unbalanced data as well.

**Comment by pabloEntropia** [comment]
Implemented in 3c2ca55 (PR #86)

However when you say,

> Make  sure cross-validation is not affected by unbalanced data


the expected behavior is not clear for me. My intuition was to compute the weights for each fold and not globally for whole the dataset as this way the SVM input will always be balanced. Do you agree?

**Comment by dbogdanov** [comment]
That is correct. However, we can also consider doing stratified splits (e.g., see [sklearn](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedShuffleSplit.html)) as curently we just randomly split all the data disregarding the distribution of labels. What do you think?
